### PR TITLE
HPCC-29199 Remove SCMInterface only ECM files

### PR DESCRIPTION
**Pull request by kenrowland**
Removed SCMInterface only ECM files.
Created hpp files for defined interfaces and added to source tree 
Updated CMake files to reflect newly added hpp interface files

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/bindings/soapesp.hpp [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+#include "jstring.hpp"
+
+interface ISoapMessage : extends IInterface
+{
+    virtual const char * getMessageType() = 0;
+    virtual StringBuffer & toString(StringBuffer & str) = 0;
+    virtual IEspContext * queryContext() = 0;
+};
+
+
```
Comment (afishbeck) [inline esp/bindings/soapesp.hpp:38]: Trivial but doesn't really need 2 blank lines.?.  I'll approve as is though.

**Review by afishbeck — APPROVED** [review]

**Review by ghalliday — CHANGES_REQUESTED** [review]
One test is failing for a related reason - not sure exactly why though.

**Comment by kenrowland** [comment]
@ghalliday @afishbeck @AttilaVamos 

The test that is failing appears to not be a completely clean build. Inspection of the output shows the following as an example of an error generated during the build

/hpcc-dev/build/generated/esp.hpp:627:11: note: previous definition of 'struct IEspNgServiceBinding'

Note that the duplicate symbol is in the 'generated' directory. This PR has removed the ECM file that generates the referenced header file (esp.hpp in the example above).

Who is responsible for the "Docker smoketest build / build-images (pull_request)" build?

**Comment by kenrowland** [comment]
https://track.hpccsystems.com/browse/HPCC-29351 created to fix the Smoketest build failure

**Comment by kenrowland** [comment]
@GordonSmith I can confirm that I can create an image with ./dockerfiles/incr.sh locally

**Review by ghalliday — APPROVED** [review]

**Pull request merged**